packages/adapter-sveltekit/src/plugin.ts

```diff
@@ -200,14 +201,16 @@ export const plugin = defineSliceMachinePlugin<PluginOptions>({
 		////////////////////////////////////////////////////////////////
 
 		hook("slice-template-library:read", async (data, context) => {
+			const isSvelte5 = await checkIsSvelte5({ helpers: context.helpers });
+
 			return await readSliceTemplateLibrary({
 				...data,
 				...context,
 				dirName: path.dirname(fileURLToPath(new URL(import.meta.url))),
 				templates: [Hero, CustomerLogos, AlternateGrid, CallToAction],
 				componentFileNames: {
-					js: "javascript.svelte",
-					ts: "typescript.svelte",
+					js: isSvelte5 ? "javascript.5.svelte" : "javascript.svelte",
```

Comment:
Related to an above comment, we can treat Svelte <5 as the exception in the file name rather than strictly 5.

We could use a descriptor rather than a version number, like `with-let-props`.